### ISSUE TITLE
Documentation missing on how to create a new module

### ISSUE THREAD

**Issue by allan-simon** [taigaio/taiga-back#565]
I haven't checked specifically in the code, but I goes with the impression that "modules"  (currently  scrum/kanban / wiki/ issues) can accept third-party one ? if so, can there be some documentation on how to create one , even if it is just at first a link to a file implementing "wiki" module?

**Comment by superalex** [comment]
You are right, we should add that piece of documentation so it's easier for users to add extra features to Taiga.

Actually you could base on existing plugins https://github.com/search?q="taiga-contrib-" to see how it could be implemented.

Regards!,

**Comment by allan-simon** [comment]
thanks, yeah right I didn't think about searching with taiga-contrib , however none of these plugins seems to integrate a new module in the sense of  wiki/issues etc

**Comment by superalex** [comment]
Yes wiki, issues, backlog...are in the taiga core by themselves but I think that using the plugin system you could create new "sections"

**Comment by darkmeiun** [comment]
what if the project is more than a mere section and needs a lot more integration in how tasks etc are handled??

**Comment by superalex** [comment]
@darkmeiun it will depend on that level of customization, if you want to change how Taiga works at a very basic level probably you will finish forking the project and making those changes. Just remember that as the project is licensed as AGPL those modifications should be published for your users.

Regards!,

**Comment by muellert** [comment]
Generally, docs on how to best develop for Taiga would be much appreciated, alongside the testing philosophy etc. I'm getting started on writing extensions, but I, and also my boss, want them to be merged upstream. Knowing what to aim at would be great!
